apps/emqx_cluster_link/src/emqx_cluster_link_router_syncer.erl

```diff
@@ -447,22 +451,21 @@ init_remote_actor(
         ),
         ClientPid
     ),
-    St1 =
-        case Res of
-            ok ->
-                St#st{status = connecting};
-            {error, Reason} ->
-                ?SLOG(error, #{
-                    msg => "cluster_link_init_failed",
-                    reason => Reason,
-                    target_cluster => TargetCluster,
-                    actor => Actor
-                }),
-                _ = maybe_alarm(Reason, St),
-                St#st{error = Reason, status = disconnected}
-        end,
-    TRef = erlang:start_timer(?ACTOR_REINIT_TIMEOUT, self(), actor_reinit),
-    St1#st{actor_init_req_id = ReqId, actor_init_timer = TRef}.
+    case Res of
+        ok ->
+            TRef = erlang:start_timer(?ACTOR_REINIT_TIMEOUT, self(), actor_reinit),
```

Comment:
It does make sense.  However, that's sounds like a much bigger refactoring, and for something nice but not strictly necessary.  Perhaps we could leave changing this to a `gen_statem` to a future PR, since this was deemed a release blocker?

Reply:
Good point.